draft-ietf-sidrops-aspa-verification.xml

```diff
@@ -310,26 +303,102 @@ authorized(AS x, AS y) =  / Else, "Provider+" if the VAP-SPAS entry
         If an attacker creates a route leak intentionally, they may try to strip their AS from the AS_PATH.
         To partly guard against that, a check is necessary to match the most recently added AS in the AS_PATH to the BGP neighbor's ASN.
         This check MUST be performed as specified in Section 6.3 of <xref target="RFC4271"/>.
-        If the check fails, then the AS_PATH is considered a Malformed AS_PATH and the UPDATE is considered to be in error (Section 6.3 of <xref target="RFC4271"/>).
+        If the check fails, then the AS_PATH is considered a Malformed AS_PATH and the UPDATE is in error (Section 6.3 of <xref target="RFC4271"/>).
         The case of transparent RS MUST also be appropriately taken care of (e.g., by suspending the neighbor ASN check).
-        The check fails also when the AS_PATH is empty (zero length) and such UPDATEs will also be considered to be in error.
+        The AS_PATH check fails also when it is empty (zero length) since such an UPDATE in eBGP is considered to be an error.
       </t>
       <t>
         <xref target="I-D.ietf-idr-deprecate-as-set-confed-set"/> specifies that "treat-as-withdraw" error handling <xref target="RFC7606"/> SHOULD be applied to routes with AS_SET in the AS_PATH.
         In the current document, routes with AS_SET are given Invalid evaluation in the AS_PATH verification procedures (<xref target="Upflow"/> and <xref target="Downflow"/>).
         See <xref target="mitigation"/> for how routes with Invalid AS_PATH are handled.
       </t>
       <t>
-        In <xref target="Upflow"/> and <xref target="Downflow"/> below, the terms "upstream path" and "downstream path" generally refer to AS paths received in the upstream direction (from a customer, lateral peer, RS-client, RS) and in the downstream direction (provider), respectively.
+        The term "upstream path" (<xref target="Upflow"/>) refers to AS paths received from a customer, or a lateral peer, or received by an RS from an RS-client, or received by an RS-client from an RS. The term "downstream path" (<xref target="Downflow"/>) refers to AS paths received from a provider or mutual-transit neighbor.
+        An AS decides if upstream or downstream path verification must be applied based on its locally configured BGP Role (<xref target="peering"/>, <xref target="role"/>) for the neighbor from which the AS path is received.
       </t>
 
-	<section title="Principles" anchor="principles">
+    <section title="Semantics of Valid, Invalid, and Unknown" anchor="semantics">
+      <t>
+        The ASPA-based path verification algorithms (<xref target="Upflow"/> and <xref target="Downflow"/>) evaluate an AS_PATH as Valid, Invalid, or Unknown.
+        'Invalid' outcome means that the AS_PATH contains a route leak (i.e., valley-free violation <xref target="RFC7908"/>) based on the available ASPAs.
```

Comment:
It just abnormal path, we don't know if it is a route leak a malformed path

Reply:
@mitradir 
Good observation. Section 7.1 is redundant and perhaps misleading because the next section "Basic Principles of Route Leak Detection" defines the terms properly anyway. I have removed Section 7.1 (what used to be "Semantics of Valid, Invalid, and Unknown").